### PR TITLE
VideoPress: add edit button to VideoDetailsCard component

### LINKED ISSUE #25816: VideoPress: Add edit button to the VideoDetailsCard
Image

### PR DESCRIPTION
**Pull request by retrofox**
This PR implements the edit button to the VideoDetailsCard component.

Fixes #

#### Changes proposed in this Pull Request:

* VideoPress: add edit button to VideoDetailsCard component. Improve story.

#### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Testing instructions:

* Use storybook
* Edit the props. Confirm the changes are propagated to the story
* Test how the component triggers the callback when clicking on the dropdown menu items

https://user-images.githubusercontent.com/77539/186491002-672a589d-64d4-49c3-9947-b22e5243e552.mov

### CODE REVIEW

File: projects/packages/videopress/changelog/update-videopress-add-edit-to-video-details-card [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Significance: patch
+Type: added
+
+
```
Comment (renatoagds) [inline projects/packages/videopress/changelog/update-videopress-add-edit-to-video-details-card]: Is missing the changelog description, not?

File: projects/js-packages/components/components/button/types.ts [hunk 2]
```diff
@@ -10,7 +10,7 @@ type JetpackButtonBaseProps = {
 	isExternalLink?: boolean;
 	size?: 'normal' | 'small';
 	text?: string;
-	variant?: 'primary' | 'secondary' | 'link';
+	variant?: 'primary' | 'secondary' | 'link' | 'tertiary';
```
Comment (renatoagds) [inline projects/js-packages/components/components/button/types.ts:13]: Based at `Jetpack Library` we should not support `tertiary`:

image

TTzDk3dzct551hYWRXPFa3-fi-2393%3A17487

File: projects/packages/videopress/src/client/admin/components/video-details-card/style.module.scss [hunk 3]
```diff
@@ -12,11 +12,28 @@
...
+		}
+	}
 }
 
 .thumbnail {
-	display: flex;
-	border-radius: var( --jp-border-radius );
+	position: relative;
+	> img {
+		display: flex;
```
Comment (renatoagds) [inline projects/packages/videopress/src/client/admin/components/video-details-card/style.module.scss]: Which is the effect of this `display: flex`? Strange to use it alone, applied to an image.

**Review by renatoagds — DISMISSED** [review]
Got some difference into the spacing from `list` and `toggle` button, and is missing the green outline on focus:

_Figma at right_

Screen Shot 2022-08-24 at 15 55 44 Screen Shot 2022-08-24 at 15 55 44

And is not possible to navigate through keyboard arrows.

Using Tabs works, but with up and down arrows, it doesn't work.

File: projects/js-packages/components/components/button/types.ts (hunk 2 above)
Comment (retrofox) [reply projects/js-packages/components/components/button/types.ts:13]: Yes, I'm aware of it. Its usage is not encourage but not forbidden either:

> ...If you feel you absolutely need it, bring it up with the team please.

I do think the `tertiary` option fits very well with what we need to implement here. Simplifies the menu implementation very well, and considering that it's a valid option for the core component, I wonder why it is not possible to use it here.

File: projects/packages/videopress/src/client/admin/components/video-details-card/style.module.scss (hunk 3 above)
Comment (retrofox) [reply projects/packages/videopress/src/client/admin/components/video-details-card/style.module.scss]: mmm probably something that I left there by mistake. Thanks, good catch. https://github.com/Automattic/jetpack/pull/25817/commits/7549934f8ea613258902480022f11f05fb2d4ba4

File: projects/js-packages/components/components/button/types.ts (hunk 2 above)
Comment (retrofox) [reply projects/js-packages/components/components/button/types.ts:13]: [After talking a little bit about it](p1661370334075449-slack-C02TQF5VAJD), I think it's ok to introduce the `tertiary` variant.

**Comment by retrofox** [comment]
> Got some difference into the spacing from `list` and `toggle` button, and is missing the green outline on focus:
> 

This is something to address at the Button component level.

> 
> And is not possible to navigate through keyboard arrows.
> Using Tabs works, but with up and down arrows, it doesn't work.

I think Tabs is the natural support for it.

**Review by oskosk — APPROVED** [review]
LGTM

**Comment by retrofox** [comment]
r251283-wpcom

**Pull request merged**